app/controlplane/pkg/biz/workflowcontract.go

```diff
@@ -346,6 +359,19 @@ func (uc *WorkflowContractUseCase) findPolicyGroup(att *schemav1.PolicyGroupAtta
 			uc.logger.Warnf("policy group '%s' skipped since it's not found or it might use an old schema version", att.GetRef())
 			return nil, nil
 		}
+		if remoteGroup.PolicyGroup != nil {
+			// validate group arguments
+			with := att.GetWith()
+			for _, input := range remoteGroup.PolicyGroup.GetSpec().GetInputs() {
+				_, ok := with[input.GetName()]
+				if !ok && input.GetRequired() {
+					return nil, NewErrValidation(fmt.Errorf("missing required input %q for group", input.GetName()))
+				}
+				if input.GetRequired() && input.GetDefault() != "" {
```

Comment:
does the validation need to happen here?

Reply:
Yes, this is one of the duplicated cases. This is during contract creation/update.